\documentclass{article}
\usepackage{graphicx}
\usepackage[margin=1in]{geometry}

% %
% \makeatletter
% \setlength{\@fptop}{0pt}
% \makeatother
% %

\begin{document}

\title{Supplementary material for ``Visualization methods for RNA-sequencing data analysis"}
\author{Lindsay Rutter}

\maketitle

\clearpage
\null
\begin{figure}[t!]
\centerline{\includegraphics[width=\columnwidth]{../MakeFigures/sbIRClusters.jpg}}
\caption{Example application of parallel coordinate plots using the iron-metabolism soybean dataset. We filtered genes with low means and/or variance, performed a hierarchical clustering analysis with a cluster size of four, and visualized the results using parallel coordinate lines. Most non-filtered genes were in Clusters 1 and 2, which both showed overexpression in one treatment and underexpression in the other treatment. The genes in Cluster 4 mostly showed messy patterns with low signal to noise ratios. Interestingly, Cluster looked similar to Cluster 2 (large values for group N and small values for group P), except for unexpectedly large values for the third replicate of group P.
\label{sbIRClusters}}
\end{figure}

\clearpage
\null
\begin{figure}[t!]
\centerline{\includegraphics[width=\columnwidth]{../MakeFigures/sbIRClusterSigSM1.jpg}}
\caption{Example of using a scatterplot matrix to assess DEG calls from a model in the iron-metabolism soybean dataset. There were 2751 significant genes in Cluster 1 after performing a hierarchical clustering analysis with a cluster size of four. These significant genes are overlaid in green over the scatterplot matrix. They follow the expected patterns of differential expression with most green points falling along the \textit{x=y} line in the scatterplots between replicates, but deviating from the \textit{x=y} line in the scatterplots between treatments. The deviation consistently demonstrates higher expression in the P group than in the N group. Hence, these green points seem to represent genes that were significantly overexpressed in the P group, which draws the same conclusion with what we derived using the parallel coordinate plots in Figure 2 of the paper. One difficulty with plotting such a large number of DEGs onto the scatterplot matrix is that overplotting can obscure our inability to determine how many DEGs are in a given location.
\label{sbIRClusterSigSM1}}
\end{figure}

\clearpage
\null
\begin{figure}[t!]
\centerline{\includegraphics[width=\columnwidth]{../MakeFigures/sbIRClusterSigSM2.jpg}}
\caption{Example of using a scatterplot matrix to assess DEG calls from a model in the iron-metabolism soybean dataset. There were 2009 significant genes in Cluster 2 after performing a hierarchical clustering analysis with a cluster size of four. These significant genes are overlaid in mustard over the scatterplot matrix. They follow the expected patterns of differential expression with most mustard points falling along the \textit{x=y} line in the scatterplots between replicates, but deviating from the \textit{x=y} line in the scatterplots between treatments. The deviation consistently demonstrates higher expression in the N group than in the P group. Hence, these mustard points seem to represent genes that were significantly overexpressed in the N group, which draws the same conclusion with what we derived using the parallel coordinate plots in Figure 2 of the paper. One difficulty with plotting such a large number of DEGs onto the scatterplot matrix is that overplotting can obscure our inability to determine how many DEGs are in a given location.
\label{sbIRClusterSigSM2}}
\end{figure}
  
\null
\begin{figure}[t!]
\centerline{\includegraphics[width=\columnwidth]{../MakeFigures/sbIRClusterSigSM3.jpg}}
\caption{Example of using a scatterplot matrix to assess DEG calls from a model in the iron-metabolism soybean dataset. There were 861 significant genes in Cluster 3 after performing a hierarchical clustering analysis with a cluster size of four. These significant genes are overlaid in pink over the scatterplot matrix. For the most part, they follow the expected patterns of differential expression with pink points falling along the \textit{x=y} line in the scatterplots between replicates, but deviating from the \textit{x=y} line in the scatterplots between treatments. The deviation consistently demonstrates higher expression in the N group than in the P group. However, the scatterplot between replicates P.1 and P.3 show slightly higher expression in P.3, and the scatterplot between replicates P.2 and P.3 also show slightly higher expression in P.3. Hence, these pink points seem to represent genes that were significantly overexpressed in the N group, but with slight inconstencies in the replicates in the P group. The parallel coordinate plots in Figure 2 of the paper showed this same conclusion and perhaps more clearly. One difficulty with plotting such a large number of DEGs onto the scatterplot matrix is that overplotting can obscure our inability to determine how many DEGs are in a given location.
\label{sbIRClusterSigSM3}}
\end{figure}  

\clearpage
\null
\begin{figure}[t!]
\centerline{\includegraphics[width=\columnwidth]{../MakeFigures/sbIRClusterSigSM4.jpg}}
\caption{Example of using a scatterplot matrix to assess DEG calls from a model in the iron-metabolism soybean dataset. There were 17 significant genes in Cluster 4 after performing a hierarchical clustering analysis with a cluster size of four. These significant genes are overlaid in coral over the scatterplot matrix. For the most part, they do not seem to follow the expected patterns of differential expression: In many of the scatterplots between treatments, the coral points do not seem to deviate much from the \textit{x=y} line. Moreover, in the scatterplots between P.1 and P.2 as well as P.1 and P.3, the coral points seems to indicate an underexpression of the P.1 replicate. We found a similar finding of somewhat messy looking DEG calls in Cluster 4 from Figure 2 in the paper. 
\label{sbIRClusterSigSM4}}
\end{figure}  

\begin{figure}[!tpb]
\centerline{\includegraphics[width=\columnwidth]{../Bioinformatics/Pictures/mdsSwitch.png}}
\caption{Boxplots and MDS plots are popular plotting tools for RNA-seq analysis. This figure shows these traditional plots applied to the cotyledon data before sample switching (left half) and after sample switching (right half). We cannot suspect from the right boxplot that samples S1.3 and S2.1 have been swapped (subplots A). This is because all six samples have similar five number summaries. For the MDS plots, we do see a cleaner separation of the two treatment groups across the first dimension in the left plot than in the right plot (subplots B). However, taking into account the second dimension, both MDS plots contain three clusters, with sample S1.1 appearing in its own cluster. Without seeing one distinct cluster for each of the two treatment groups, it is difficult to suspect that samples S1.3 and S2.1 have been swapped in the right MDS plot (subplots B). We can only derive clear suspicion that the samples may have been switched by using plots that provide gene-level resolution like with the scatterplot matrix (Figure~\ref{sbCNSwitchedSM}).
\label{mdsSwitch}}
\end{figure}

\clearpage
\null 
\begin{figure}[t!]
\centerline{\includegraphics[width=0.7\columnwidth]{../MakeFigures/Dashboards/litreCluster1/litreCluster1.jpg}}
\caption{Litre plots for significant genes inside Cluster 1 from Figure 2 of the paper. Subplots A and B each overlay a significant gene from Cluster 1 as nine green points. The genes show a pattern expected of a differentially-expressed one, by clumping together and deviating from the \textit{x=y} line. Moreover, the genes appear over-expressed in the P group. This is consistent with what we saw in Figure 2 of the paper. To interactively view the litre plot for all significant genes within Cluster 1, please visit https://rnaseqvisualization.shinyapps.io/litreCluster1.
\label{litreCluster1}}
\end{figure}   

\null
\begin{figure}[t!]
\centerline{\includegraphics[width=0.65\columnwidth]{../MakeFigures/lkClustersKeep.jpg}}
\caption{Parallel coordinate plots showing hiearchical clustering analysis results of size eight for the 3,974 genes that remained in the kidney-specific DEGs after TMM normalization. We see that, for the most part, the parallel coordinate patterns follow the expected patterns across the clusters. The ideal pattern of DEGs is especially captured in the first cluster (the largest one with 1,136 genes). We used hierarchical clustering to mitigate additional overplotting that would occur if we were to plot all genes onto only one parallel coordinate plot.
\label{lkClustersKeep}}
\end{figure}

\null
\begin{figure}[t!]
\centerline{\includegraphics[width=0.65\columnwidth]{../MakeFigures/lkClustersOrig.jpg}}
\caption{Parallel coordinate plots showing hiearchical clustering analysis results of size eight for the 1,968 genes that were initially designated liver-specific DEGs after library scale normalization. We see that, for the most part, the parallel coordinate patterns follow the expected patterns across the clusters. The ideal pattern of DEGs is especially captured in the first cluster (the largest one with 933 genes). We used hierarchical clustering to mitigate additional overplotting that would occur if we were to plot all genes onto only one parallel coordinate plot.
\label{lkClustersOrig}}
\end{figure}

\null
\begin{figure}[t!]
\centerline{\includegraphics[width=0.65\columnwidth]{../MakeFigures/lkClustersRemove.jpg}}
\caption{Parallel coordinate plots showing hiearchical clustering analysis results of size eight for the 3,076 genes that were removed from the kidney-specific DEGs after TMM normalization. Unlike in Figure~\ref{KeepPCP}, the patterns in almost all clusters do not resemble the expected DEG format; instead, they show large variability between replicates and small variability between groups. In some clusters, it is difficult to even determine which group would be the overexpressed one if its genes were in fact DEGs. Taken together, this plot provides additional statistical evidence that the application of TMM normalization successfully removed genes that were previously mislabeled as kidney-specific DEGs (in Figure~\ref{RawKLSig}) with library scaling normalization. We used hierarchical clustering to mitigate additional overplotting that would occur if we were to plot all genes onto only one parallel coordinate plot.
\label{lkClustersRemove}}
\end{figure}

\null
\begin{figure}[t!]
\centerline{\includegraphics[width=0.65\columnwidth]{../MakeFigures/lkClustersAdd.jpg}}
\caption{Parallel coordinate plots showing hiearchical clustering analysis results of size eight for the 1,578 genes that were \textit{added} as liver-specific DEGs after TMM normalization. We see that, for the most part, the parallel coordinate patterns follow the expected patterns across the clusters. We used hierarchical clustering to mitigate additional overplotting that would occur if we were to plot all genes onto only one parallel coordinate plot.
\label{lkClustersAdd}}
\end{figure}

\null
\begin{figure}[t!]
\centerline{\includegraphics[width=1\columnwidth]{../MakeFigures/lkClustersKeepSM.jpg}}
\caption{Scatterplot matrix of the 1,136 genes that were in the first cluster (of Figure~\ref{lkClustersKeep}) from genes that remained as kidney-specific DEGs even after TMM normalization. With this scatterplot matrix, we verify from an additional perspective that these genes demonstrate the expected patterns of DEGs.
\label{lkClustersKeepSM}}
\end{figure}

\null
\begin{figure}[t!]
\centerline{\includegraphics[width=1\columnwidth]{../MakeFigures/lkClustersOrigSM.jpg}}
\caption{Scatterplot matrix of the 933 genes that were in the first cluster (of Figure~\ref{lkClustersOrig}) from genes that were initially designated as liver-specific DEGs after library scale normalization. With this scatterplot matrix, we verify from an additional perspective that these genes demonstrate the expected patterns of DEGs.
\label{lkClustersOrigSM}}
\end{figure}

\null
\begin{figure}[t!]
\centerline{\includegraphics[width=1\columnwidth]{../MakeFigures/lkClustersRemoveSM.jpg}}
\caption{Scatterplot matrix of the 529 genes that were in the first cluster (of Figure~\ref{lkClustersRemove}) from genes that no longer remained as kidney-specific DEGs after TMM normalization. With this scatterplot matrix, we verify from an additional perspective that these genes do not demonstrate the expected patterns of DEGs too strongly (they do not deviate much from the \textit{x=y} line in the treatment scatterplots). This provides additional evidence that TMM normalization removing these genes from DEG status may be valid.
\label{lkClustersRemoveSM}}
\end{figure}

\null
\begin{figure}[t!]
\centerline{\includegraphics[width=1\columnwidth]{../MakeFigures/lkClustersAddSM.jpg}}
\caption{Scatterplot matrix of the 317 genes that were in the first cluster (of Figure~\ref{lkClustersAdd}) from genes that were \textit{added} as liver-specific DEGs after TMM normalization. With this scatterplot matrix, we are \textbf{unable} to verify from an additional perspective that these genes demonstrate the expected patterns of DEGs. (Need to find an explanation for this).
\label{lkClustersAddSM}}
\end{figure}


\null
\begin{figure}[t!]
\centerline{\includegraphics[width=0.7\columnwidth]{../MakeFigures/Dashboards/litreClusterKeep/litreClusterKeep.jpg}}
\caption{Example litre plots from the 1,136 genes that were in the first cluster (of Figure~\ref{lkClustersKeep}) of genes that remained kidney-specific DEGs even after TMM normalization. With these litre plots, we verify from an additional perspective that these genes demonstrate the expected patterns of DEGs.
\label{litreClusterKeep}}
\end{figure}

\null
\begin{figure}[t!]
\centerline{\includegraphics[width=0.7\columnwidth]{../MakeFigures/Dashboards/litreClusterOrig/litreClusterOrig.jpg}}
\caption{Example litre plots from the 933 genes that were in the first cluster (of Figure~\ref{lkClustersOrig}) from genes that were initially designated as liver-specific DEGs after library scale normalization. With these litre plots, we verify from an additional perspective that these genes demonstrate the expected patterns of DEGs.
\label{litreClusterOrig}}
\end{figure}

\null
\begin{figure}[t!]
\centerline{\includegraphics[width=0.7\columnwidth]{../MakeFigures/Dashboards/litreClusterRemove/litreClusterRemove.jpg}}
\caption{Example litre plots from the 529 genes that were in the first cluster (of Figure~\ref{lkClustersRemove}) of genes that no longer remained as kidney-specific DEGs after TMM normalization. With these litre plots, we verify from an additional perspective that these genes do not demonstrate the expected patterns of DEGs. This provides additional evidence that TMM normalization removing these genes from DEG status may be valid.
\label{litreClusterRemove}}
\end{figure}

\null
\begin{figure}[t!]
\centerline{\includegraphics[width=0.7\columnwidth]{../MakeFigures/Dashboards/litreClusterAdd/litreClusterAdd.jpg}}
\caption{Example litre plots from the 1,578 genes that were in the first cluster (of Figure~\ref{lkClustersAdd}) from genes that were \textit{added} as liver-specific DEGs after TMM normalization. With these litre plots, we are \textbf{unable} to verify from an additional perspective that these genes demonstrate the expected patterns of DEGs. (Need to find an explanation for this).
\label{litreClusterAdd}}
\end{figure}




\null
\begin{figure}[t!]
\centerline{\includegraphics[width=1\columnwidth]{../MakeFigures/lkClustersKeepSM-St.jpg}}
\caption{Scatterplot matrix of the \textit{standardized} 1,136 genes that were in the first cluster (of Figure~\ref{lkClustersKeep}) from genes that remained as kidney-specific DEGs even after TMM normalization. Even though the standardization process removes the interesting geometrical features we saw back in Figure~\ref{lkClustersKeepSM} regarding the streak of overexpressed liver genes, it allows us to view the DEG patterns in a more exaggerated fashion.
\label{lkClustersKeepSM-St}}
\end{figure}

\null
\begin{figure}[t!]
\centerline{\includegraphics[width=1\columnwidth]{../MakeFigures/lkClustersOrigSM-St.jpg}}
\caption{Scatterplot matrix of the \textit{standardized} 933 genes that were in the first cluster (of Figure~\ref{lkClustersOrig}) from genes that were initially designated as liver-specific DEGs after library scale normalization. Even though the standardization process removes the interesting geometrical features we saw back in Figure~\ref{lkClustersKeepSM} regarding the streak of overexpressed liver genes, it allows us to view the DEG patterns in a more exaggerated fashion.
\label{lkClustersOrigSM-St}}
\end{figure}

\null
\begin{figure}[t!]
\centerline{\includegraphics[width=1\columnwidth]{../MakeFigures/lkClustersRemoveSM-St.jpg}}
\caption{Scatterplot matrix of the \textit{standardized} 529 genes that were in the first cluster (of Figure~\ref{lkClustersRemove}) from genes that no longer remained as kidney-specific DEGs after TMM normalization. Even though the standardization process removes the interesting geometrical features we saw back in Figure~\ref{lkClustersKeepSM} regarding the streak of overexpressed liver genes, it allows us to view the DEG patterns in a more exaggerated fashion.
\label{lkClustersRemoveSM-St}}
\end{figure}

\null
\begin{figure}[t!]
\centerline{\includegraphics[width=1\columnwidth]{../MakeFigures/lkClustersAddSM-St.jpg}}
\caption{Scatterplot matrix of the \textit{standardized} 317 genes that were in the first cluster (of Figure~\ref{lkClustersAdd}) from genes that were \textit{added} as liver-specific DEGs after TMM normalization. Even though the standardization process removes the interesting geometrical features we saw back in Figure~\ref{lkClustersKeepSM} regarding the streak of overexpressed liver genes, it allows us to view the DEG patterns in a more exaggerated fashion.
\label{lkClustersAddSM-St}}
\end{figure}







\end{document}