\documentclass[12pt]{article}
\usepackage[margin=1in]{geometry}
\usepackage{mathptmx}
\usepackage[scaled=.90]{helvet}
\usepackage{setspace}

\begin{document}

\noindent
\textbf{Title}: \break
Visualization methods for RNA-sequencing data analysis
\vspace{5mm} \break
\noindent{\textbf{Authors}}: \break
Lindsay Rutter (lrutter@iastate.edu) and Dianne Cook (dicook@monash.edu)
\vspace{5mm} \break
\noindent{\textbf{Abstract}}: \break
Numerous studies have shown that RNA-seq data is replete with biases and accurate detection of differentially expressed genes is not a trivial task. In light of these complications, researchers should analyze RNA-seq data like they would any other biased multivariate data. The most effective approach to modern data analysis is to iterate between models and visuals, and to enhance the appropriateness of applied models based on feedback from visuals. Unfortunately, researchers do not commonly use models and visuals in a complimentary fashion when analyzing RNA-seq data. Here, we use real RNA-seq data to demonstrate that our plotting tools can detect normalization problems, DEG designation problems, and common errors in the RNA-seq analysis pipeline. In this paper, we do not propose that users radically change their approach to RNA-seq analysis. Instead, we propose that users simply modify their approach to RNA-seq analysis by assessing their models with multivariate graphical tools.
\vspace{5mm} \break
\noindent{\textbf{Keywords}}: \break
Data visualization; Exploratory data analysis; Interactive statistical graphics; RNA-sequencing

\end{document}
